Build fails on Android with latest NDK

When building SDL 2.0.22 on Android with the latest NDK, I'm seeing the following error:

```
/home/vsts/work/1/s/build/SDL2-2.0.22/src/core/android/SDL_android.c:515:13: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
    JNIEnv *env = NULL;
            ^
/home/vsts/work/1/s/build/SDL2-2.0.22/src/core/android/SDL_android.c:2564:13: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
    jstring jpermission = (*env)->NewStringUTF(env, permission);
```

It looks like this warning is explicitly enabled in the root CMakeLists.txt if the compiler supports it. I'm wondering if it wasn't supported until a recent release.

@slouken unfortunately I'm just getting another error later in the build. This also happens with the snapshot you linked. This is due to including the cpu-features.c file from the NDK directly into the project while applying the flags used for normal SDL code. The errors are:

```
/usr/local/lib/android/sdk/ndk/25.0.8775105/sources/android/cpufeatures/cpu-features.c:514:5: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
    struct { uint32_t tag; uint32_t value; } entry;
    ^
/usr/local/lib/android/sdk/ndk/25.0.8775105/sources/android/cpufeatures/cpu-features.c:559:11: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
    char* cpuFeatures = extract_cpuinfo_field(cpuinfo, cpuinfo_len, "Features");
          ^
/usr/local/lib/android/sdk/ndk/25.0.8775105/sources/android/cpufeatures/cpu-features.c:848:19: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
            char* value_end = value + strlen(value);
                  ^
/usr/local/lib/android/sdk/ndk/25.0.8775105/sources/android/cpufeatures/cpu-features.c:729:18: error: mixing declarations and code is incompatible with standards before C99 [-Werror,-Wdeclaration-after-statement]
        uint32_t hwcaps = getauxval(AT_HWCAP);
                 ^
```

You can see the upstream file here: https://android.googlesource.com/platform/ndk/+/master/sources/android/cpufeatures/cpu-features.c

For now I'll probably patch out the block setting werror for declaration after code.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -600,13 +600,17 @@ if(USE_GCC OR USE_CLANG)
     list(APPEND EXTRA_CFLAGS "-fno-strict-aliasing")
   endif()
 
-  check_c_compiler_flag(-Wdeclaration-after-statement HAVE_GCC_WDECLARATION_AFTER_STATEMENT)
-  if(HAVE_GCC_WDECLARATION_AFTER_STATEMENT)
-    check_c_compiler_flag(-Werror=declaration-after-statement HAVE_GCC_WERROR_DECLARATION_AFTER_STATEMENT)
-    if(HAVE_GCC_WERROR_DECLARATION_AFTER_STATEMENT)
-      list(APPEND EXTRA_CFLAGS "-Werror=declaration-after-statement")
+  # Android needs to bring in external files that don't adhere to the declaration-after-statement
+  # warning, so skip this warning on Android.
+  if(NOT ANDROID)
+    check_c_compiler_flag(-Wdeclaration-after-statement HAVE_GCC_WDECLARATION_AFTER_STATEMENT)
+    if(HAVE_GCC_WDECLARATION_AFTER_STATEMENT)
+      check_c_compiler_flag(-Werror=declaration-after-statement HAVE_GCC_WERROR_DECLARATION_AFTER_STATEMENT)
+      if(HAVE_GCC_WERROR_DECLARATION_AFTER_STATEMENT)
+        list(APPEND EXTRA_CFLAGS "-Werror=declaration-after-statement")
+      endif()
+      list(APPEND EXTRA_CFLAGS "-Wdeclaration-after-statement")
     endif()
-    list(APPEND EXTRA_CFLAGS "-Wdeclaration-after-statement")
   endif()
 
   if(DEPENDENCY_TRACKING)

